pkg/network/server_test.go

```diff
@@ -135,17 +137,29 @@ func TestServerRegisterPeer(t *testing.T) {
 
 	startWithCleanup(t, s)
 
-	s.register <- ps[0]
-	require.Eventually(t, func() bool { return 1 == s.PeerCount() }, time.Second, time.Millisecond*10)
-	s.handshake <- ps[0]
+	var wg sync.WaitGroup
+
+	// Register and handshake peers
+	for i, p := range ps {
+		wg.Add(1)
```

Comment:
I don't think that this WG-related code fixes the problem. As you mentioned, the problem is likely related to the fact that zap logger writes logs after the test ends. The reason is likely connected with https://github.com/uber-go/zap/issues/687, so the https://github.com/uber-go/zap/issues/687#issuecomment-473413884 might help to fix it.

To prevent logs after the test ends we need to properly shutdown the server. For this we have a call to `startWithCleanup(t, s)` at the start of the test, it has the following call:
```
t.Cleanup(func() {
		s.Shutdown()
	})
```
So I think we need to check whether this "Cleanup" allows log inside it and whether the https://github.com/uber-go/zap/issues/687 is relevant for the "Cleanup" functions. Maybe need to look inside the https://github.com/golang/go/blob/master/src/testing/testing.go. @roman-khimov, need you here, what do you think?

Reply:
So in other words I think that this problem is related to conflict between `zap` and cleanup testing functionality. And the problem may be fixed by rolling back from `startWithCleanup(t, s)` to `defer s.Shutdown()`.